aep/general/0231/aep.md.j2

```diff
@@ -1,5 +1,182 @@
+  collection in the path of any resource being retrieved does not match, the
+  request **must** fail.
+  - This field **should** be required if only 1 parent per request is allowed.
+  - The field **should** identify the [resource type][aep-122-parent] that it
+    references.
+  - The comment for the field **should** document the resource pattern.
+- There **must not** be a body key in the `google.api.http` annotation.
+
+{% tab oas %}
+
+{% sample 'batchget.oas.yaml', 'paths' %}
+
+- The `paths` parameter **must** be a query parameter which accepts an array of
+  resource paths specifying the resources to retrieve. The parameter **should**
+  be named `paths`.
+  - If no resource paths are provided, the API **should** error with status
+    code `400 Bad Request`.
+  - If the collection in the path of any resource does not match the collection
+    of the request URL, the request **must** fail.
+- The method definition **must not** have a `requestBody` defined.
+
+{% endtabs %}
+
+### Response
+
+The response for a batch get method **should** be specified with the following
+pattern:
+
+- The response schema **must** match the method name, with `Response` suffix.
+- The response schema **must** have a single array property where each item is
+  either a retrieved resource or an error structure describing an error that
+  occurred attempting to retrieve the resource. The error alternative would
+  only be present for non-transactional batch gets.
+- The order of resources/error objects in the response **must** be the same as
+  the paths in the request.
+- The array of resources **must** be named `results` and contain resources with
+  no additional wrapping.
+- There must not be a `nextPageToken` field as batch get operations are not
+  pageable.
+
+{% tab proto -%}
+
+```proto
+message BatchGetBooksResponse {
+  // Books requested.
+  repeated Book books = 1;
+}
+```
+
+{% tab oas %}
+
+{% sample 'batchget.oas.yaml', 'paths./publishers/{publisherId}/books:BatchGet.get.responses.200.content.application/json' %}
+
+Example response body:
+
+```json
+{
+  "results": [
+    {
+      "name": "publishers/lacroix/books/les-mis",
+      "isbn": "978-037-540317-0",
+      "title": "Les Misérables",
+      "authors": ["Victor Hugo"],
+      "rating": 9.6
+    },
+    {
+      "type": "https://datatracker.ietf.org/doc/html/rfc9110#name-404-not-found",
+      "title": "The resource is not available"
+    }
+    {
+      "name": "publishers/lacroix/books/hunchback-of-notre-dame",
+      "isbn": "978-140-274575-1",
+      "title": "The Hunchback of Notre Dame",
+      "authors": ["Victor Hugo"],
+      "rating": 9.3
+    }
+  ]
+}
+```
+
+{% endtabs %}
+
+### Support for transactional get
+
+The batch get method may support a transactional form of operation where the
+get either succeeds for all requested resources or fails. When supported, the
+method should define a boolean parameter `transactional` that the user must
```

Comment:
```suggestion
method must define a boolean parameter `transactional` that the user must
```

If we don't require that - it might be difficult for clients to understand the purpose of the value and use it.

Or we could require an annotation to specify this is the "transactional" parameter?